src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -706,6 +706,21 @@ public function setRelation($relation, $value)
         return $this;
     }
 
+    /**
+     * Unset an existing relation.
+     *
+     * @param  string  $relation
+     * @return $this
+     */
+    public function unsetRelation($relation)
+    {
+        if ($this->relationLoaded($relation)) {
```

Comment:
Is this necessary? As long as $this->relations isn't null, `unset($this->relations('foo'))` will always work

Reply:
You are right, thought it could cause an undefined index error, will change